File: 0033-搜索旋转排序数组/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        if(nums.length < 1){
            return -1;
        }
        if(nums[0] == target){
            return 0;
        }else if(nums[0] < target){
            for(int i=1; i<nums.length && nums[i-1] <= nums[i]; i++){
                if(target == nums[i]){
                    return i;
                }
            }
        }else{
            int i = 1;
            while(i<nums.length && nums[i-1] <= nums[i]){ i++; }
            if(i < nums.length){
                for(; i<nums.length; i++){
                    if(target == nums[i]){
                        return i;
                    }
                }
            }
        }
        return -1;
    }
}

File: 0033-搜索旋转排序数组/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums)-1
        while i <= j:
            mid = (i + j) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] < nums[j]:
                if target > nums[mid] and target <= nums[j]:
                    i = mid + 1
                else:
                    j = mid - 1
            else:
                # nums[mid] > nums[j]
                if target >= nums[i] and target < nums[mid]:
                    j = mid - 1
                else:
                    i = mid + 1
        return -1
